### PR TITLE
Fix relationships 

### PR DESCRIPTION
**Pull request by rafaeljesus**
This PR implements a new field `match`  which allows seamless dumping table based on the matching condition.

Here are some example of the matching conditions:
- `Match = "users.id IN (1, 2, 3)"`
- `Match = "users.id >= 500`
- `Match = "users.created_at BETWEEN '2018-01-01' AND now()`

The users -> orders sample could be mapped as follows:
```toml
[[Tables]]
  Name = "users"
  [Tables.Filter]
    Match = "users.created_at BETWEEN '2017-01-01' AND now()"

[[Tables]]
  Name = "orders"
  [Tables.Filter]
    Match = "users.created_at BETWEEN '2017-01-01' AND now()"
  [[Tables.Relationships]]
    ReferencedTable = "users"
    ReferencedKey = "id"
    ForeignKey = "user_id"
```

### CODE REVIEW

**Comment by mereba** [comment]
@rafaeljesus this is so great! Can we update `init` to generate the right config?

File: pkg/anonymiser/anonymiser.go [hunk 1]
```diff
@@ -25,26 +33,25 @@ func NewAnonymiser(source reader.Reader, tables config.Tables) reader.Reader {
...
+// ReadTable wraps reader.ReadTable method for anonymising rows published from the reader.Reader
+func (a *anonymiser) ReadTable(tableName string, rowChan chan<- database.Row, opts reader.ReadTableOpt, configTables config.Tables) error {
 	logger := log.WithField("table", tableName)
-
 	logger.Debug("Loading anonymiser config")
 	table, err := a.tables.FindByName(tableName)
 	if err != nil {
 		logger.WithError(err).Debug("the table is not configured to be anonymised")
-		return a.Reader.ReadTable(tableName, rowChan, opts)
+		return a.Reader.ReadTable(tableName, rowChan, opts, configTables)
```
Comment (boekkooi-fresh) [inline pkg/anonymiser/anonymiser.go]: Maybe we should start thinking about a context?

Comment (rafaeljesus) [reply pkg/anonymiser/anonymiser.go]: This came to my my mind for managing query timeouts, however I still didn't find what would be a proper timeout value for reading and writing, that's why I didn't implemented. Is that what you mean?

Comment (rafaeljesus) [reply pkg/anonymiser/anonymiser.go]: @boekkooi-fresh let's add this as a next task so I can move my ticket ^^ wdyt?

**Comment by rafaeljesus** [comment]
@mereba I've update the init command to generate the following example ^^

```toml
[[Tables]]
  Name = "users"
  [Tables.Filter]
    Match = "users.active = TRUE"
    Limit = 100
    [Tables.Filter.Sorts]
      orderNr = "asc"
  [Tables.Anonymise]
    email = "EmailAddress"
    firstName = "FirstName"

[[Tables]]
  Name = "orders"
  [Tables.Filter]
    Match = "users.active = TRUE"

  [[Tables.Relationships]]
    ReferencedTable = "users"
    ReferencedKey = "id"
    ForeignKey = "user_id"
```

**Comment by mereba** [comment]
+1

**Comment by mereba** [comment]
:shipit:

File: pkg/dumper/generic/sql.go [hunk 2]
```diff
@@ -40,12 +40,12 @@ func NewSqlDumper(rdr reader.Reader, engine SqlEngine) dumper.Dumper {
 	}
 }
 
-func (p *sqlDumper) Dump(done chan<- struct{}, configTables config.Tables) error {
+func (p *sqlDumper) Dump(done chan<- struct{}, configTables config.Tables, concurrency int) error {
```
Comment (boekkooi-fresh) [inline pkg/dumper/generic/sql.go]: Should `concurrency` not be provided to `NewSqlDumper`?

**Pull request merged**